core/src/main/kotlin/org/evomaster/core/search/service/monitor/InterfaceAdapter.kt

```diff
@@ -0,0 +1,170 @@
+package org.evomaster.core.search.service
+
+import com.google.gson.*
+import io.swagger.util.Json
+import org.evomaster.core.problem.rest.RestCallAction
+import org.evomaster.core.search.Action
+import org.evomaster.core.search.gene.Gene
+import java.lang.reflect.Constructor
+import java.lang.reflect.Parameter
+
+import java.lang.reflect.Type
+import kotlin.reflect.KProperty1
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.allSuperclasses
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.full.isSubclassOf
+import kotlin.reflect.jvm.javaType
+import kotlin.reflect.jvm.jvmErasure
+
+/**
+ * @author: manzhang
+ * @date: 2018/9/11
+ */
+
+class InterfaceAdapter<T : Any> : JsonSerializer<T>, JsonDeserializer<T> {
+
+
+
+    companion object {
+        private val CLASSNAME = "CLASSNAME"
+        private val DATA = "DATA"
+
+        private var fake : JsonObject? = null
+
+        private fun getFakeJson() : JsonObject{
+            var obj = JsonObject()
```

Comment:
"val"  instead of "var?"
furthermore, "obj" calls can be simplified with ".apply{}"

Reply:
I modified this accordingly. But there is an issue to save ArrayGene with Genetic type, then I made this temporal solution to save it as StringGene.